querybook/webapp/components/QueryComposer/QueryComposer.tsx

```diff
@@ -715,6 +722,12 @@ const QueryComposer: React.FC = () => {
                             Object.keys(samplingTables).length > 0
                         }
                         sampleRate={getSampleRate()}
+                        onUpdateQuery={(query: string, run?: boolean) => {
+                            setQuery(query);
+                            if (run) {
+                                handleRunQuery();
```

Comment:
is handleQuery going to run with the updated state? this might create a race condition